src/aabb.rs

```diff
@@ -0,0 +1,175 @@
+    pub fn union_point(&self, other: &Point3<f32>) -> AABB {
+        AABB::new(Point3::new(self.min.x.min(other.x),
+                              self.min.y.min(other.y),
+                              self.min.z.min(other.z)),
+                  Point3::new(self.max.x.max(other.x),
+                              self.max.y.max(other.y),
+                              self.max.z.max(other.z)))
+    }
+
+    /// Returns a new minimal `AABB` which contains both this `AABB` and `other`.
+    pub fn union_bounded<T: Bounded>(&self, other: &T) -> AABB {
+        self.union_aabb(&other.aabb())
+    }
+
+    /// Returns the size of this `AABB` in all three dimensions.
+    pub fn size(&self) -> Vector3<f32> {
+        self.max - self.min
+    }
+
+    /// Returns the center point of the `AABB`.
+    pub fn center(&self) -> Point3<f32> {
+        self.min + (self.size() / 2.0)
+    }
+
+    /// Returns the total surface area of this `AABB`.
+    pub fn surface_area(&self) -> f32 {
+        let size = self.size();
+        2.0 * size.x * size.y + size.x * size.z + size.y * size.z
+    }
+}
+
+/// Make `AABB`s indexable. `aabb[0]` gives a reference to the minimum bound.
+/// All other indices return a reference to the maximum bound.
+impl Index<usize> for AABB {
+    type Output = Point3<f32>;
+
+    fn index(&self, index: usize) -> &Point3<f32> {
+        if index == 0 { &self.min } else { &self.max }
+    }
+}
+
+/// Implementation of `Bounded` for single points.
+impl Bounded for Point3<f32> {
+    fn aabb(&self) -> AABB {
+        AABB::new(*self, *self)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use aabb::AABB;
+    use nalgebra::Point3;
+
+    type TupleVec = (f32, f32, f32);
+
+    fn tuple_to_point(tpl: &TupleVec) -> Point3<f32> {
+        Point3::new(tpl.0, tpl.1, tpl.2)
+    }
+
+    /// Test whether an empty `AABB` does not contains anything.
+    quickcheck!{
+        fn test_empty_contains_nothing(tpl: TupleVec) -> bool {
+            let p = tuple_to_point(&tpl);
+            let aabb = AABB::empty();
+            !aabb.contains(&p)
+        }
+    }
+
+    /// Test whether an AABB always contains its center.
+    quickcheck!{
+        fn test_aabb_contains_center(a: TupleVec, b: TupleVec) -> bool {
+            let p1 = tuple_to_point(&a);
+            let p2 = tuple_to_point(&b);
+            let aabb = AABB::empty().union_point(&p1).union_bounded(&p2);
+            aabb.contains(&aabb.center())
+        }
+    }
+
+    /// Test whether the union of two point-sets contains all the points.
+    quickcheck!{
+        fn test_join_two_aabbs(a: (TupleVec, TupleVec, TupleVec, TupleVec, TupleVec),
+                               b: (TupleVec, TupleVec, TupleVec, TupleVec, TupleVec))
+                               -> bool {
+            let points = [a.0, a.1, a.2, a.3, a.4, b.0, b.1, b.2, b.3, b.4];
+            let points = points.iter().map(tuple_to_point).collect::<Vec<Point3<f32>>>();
+            let aabb1 =
```

Comment:
You should walk the reader through what you are doing here in general and in other tests. I imagine a form like this:
`// Set up two empty AABBs.`
`// Do this and compare that.`

Don't do this in too great detail, obviously, but I think these tests are very terse to read and 2-3 comments would ease them up a bit.